### PR TITLE
Revert redis-test versioning changes

### PR DESCRIPTION
**Pull request by jaymell**
Publishing to crates.io without the version specification 
is currently not possible. Reverting these changes for now 
until we have a better solution/workaround.

### CODE REVIEW

**Comment by jaymell** [comment]
@nihohit I'll let you review this one before merging. I think it probably makes sense to leave the version specification in (as, for example, tokio does for its test crate), though perhaps we don't need to update it for minor releases of the `redis` crate. 

For reference https://github.com/rust-lang/cargo/issues/11133

**Comment by nihohit** [comment]
LGTM. Will keeping both the version and the path arguments be an issue? Might this fall if we don't update the version correctly in both places (which might be good verification)?

**Review by nihohit — APPROVED** [review]

**Comment by jaymell** [comment]
> LGTM. Will keeping both the version and the path arguments be an issue? Might this fall if we don't update the version correctly in both places (which might be good verification)?

We should be ok as long as the version specified in the test crate is part of the same major release as the redis crate. For example, specifying a redis version of 0.23.0 would also work in the current PR, but builds will fail if it's 0.22.0 or 0.24.0. So, we should generally have warning if we forget to update the versions.

**Pull request merged**
